core/modules/comment/comment.module

```diff
@@ -733,10 +733,28 @@ function comment_node_page_additions(Node $node) {
     $comments_per_page = $node_type->settings['comment_per_page'];
     if ($cids = comment_get_thread($node, $mode, $comments_per_page)) {
       $comments = comment_load_multiple($cids);
-      comment_prepare_thread($comments);
-      $build = comment_view_multiple($comments, $node);
-      $build['pager']['#theme'] = 'pager';
-      $additions['comments'] = $build;
+      // In rare cases $comments is an empty array,
+      // for instance if the only comment entity attached to this node
+      // is broken for some reason.
+      if (!empty($comments)) {
+        comment_prepare_thread($comments);
+        $build = comment_view_multiple($comments, $node);
+        $build['pager']['#theme'] = 'pager';
+        $additions['comments'] = $build;
+      }
+      elseif ($node->comment_count > 0) {
+        // Comment count has got out of line, update it.
```

Comment:
Thanks for adding a period on this code comment, I note that the original D7 patch did not include it: https://www.drupal.org/files/issues/comment-author-deleted-1451072-2-D7.patch Thanks!